### PR TITLE
Upgrade to Ruby 3.1.0

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Upgrade to ruby 3.1
- Apply rubocop formatting

### CODE REVIEW

**Comment by iHiD** [comment]
FYI, this one is painful for me to deploy as I have to rebuild all the tooling invoker fleet of machines (as they're not docker)

**Comment by ErikSchierboom** [comment]
@iHiD I'm fine with us not doing it, as it is such as small codebase anyway, but then we should close this PR I think.

File: Gemfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ gem 'rest-client'
 gem 'exercism-config', '>= 0.76.0'
 
 gem 'rubyzip'
-gem 'rugged' # Git
+gem 'rugged', '1.4.2' # Git
```
Comment (ErikSchierboom) [inline Gemfile:16]: Should we do the same for the website? To prevent us from forgetting that we can't just upgrade it.

Comment (iHiD) [reply Gemfile:16]: Yes please.

Comment (ErikSchierboom) [reply Gemfile:16]: [PR opened](https://github.com/exercism/website/pull/2914/)

**Comment by ErikSchierboom** [comment]
Superseded by #64

**Pull request closed without merging**